code/modules/unit_tests/metabolizing.dm

```diff
@@ -34,3 +34,63 @@
 
 	TEST_ASSERT(!user.reagents.has_reagent(meth), "User still has meth in their system when it should've finished metabolizing")
 	TEST_ASSERT(!user.has_movespeed_modifier(/datum/movespeed_modifier/reagent/methamphetamine), "User still has movespeed modifier despite not containing any more meth")
+
+/datum/unit_test/addictions/Run()
+	var/mob/living/carbon/human/pill_user = allocate(/mob/living/carbon/human)
+	var/mob/living/carbon/human/syringe_user = allocate(/mob/living/carbon/human)
+	var/mob/living/carbon/human/pill_syringe_user = allocate(/mob/living/carbon/human)
+
+	var/obj/item/reagent_containers/pill/pill = allocate(/obj/item/reagent_containers/pill)
+	var/obj/item/reagent_containers/pill/pill_two = allocate(/obj/item/reagent_containers/pill)
+
+	var/obj/item/reagent_containers/syringe/syringe = allocate(/obj/item/reagent_containers/syringe)
+
+	var/datum/reagent/drug/methamphetamine/meth = allocate(/datum/reagent/drug/methamphetamine)
```

Comment:
Aren't reagents stored as singletons somewhere? Do we need a create a new instance?

Reply:
I only created a new instance because the TEST_ASSERT code is the same as the code used in metabolism:
```
var/is_addicted_to = addiction_list && is_type_in_list(R, addiction_list)
```

And is_type_in_list actually expects an atom, not a path. I could add a helper proc for is_path_in_list and then I won't need to create a new instance anymore and can go back to using initial() values?